core/src/main/java/org/zenoss/zep/impl/EventFanOutPlugin.java

```diff
@@ -42,14 +45,35 @@ public void setAmqpConnectionManager(AmqpConnectionManager amqpConnectionManager
     }
 
     @Override
+    public void startBatch(EventPostIndexContext context) throws Exception {
+        context.setPluginState(this, new ArrayList<EventSummary>(context.getIndexLimit()));
```

Comment:
Open to suggestions on sizing and the appropriate data structure to use here. I think avoiding reallocations on the indexing thread is more important than the RAM used by allocating enough space for `limit*sizeof(EventSummary)`.